src/Core/Compiler/NodeConverter.php

```diff
@@ -107,7 +107,12 @@ public function convert(NodeInterface $node)
     protected function convertEscapingNode(EscapingNode $node)
     {
         $configuration = $this->convert($node->getNode());
-        $configuration['execution'] = sprintf('htmlspecialchars(%s, ENT_QUOTES)', $configuration['execution']);
+        $configuration['execution'] = sprintf(
+            'call_user_func_array( function ($var) { ' .
+            'return (is_string($var) || (is_object($var) && method_exists($var, \'__toString\')) ' .
```

Comment:
Oh ah yeah... well... thats true. No potential security issue. And we'll get always a string as result which may be antoher problem. So yes - ignore my commts